src/functionalTest/groovy/com/autonomousapps/jvm/JavaToolchainSpec.groovy

```diff
@@ -0,0 +1,39 @@
+package com.autonomousapps.jvm
+
+import com.autonomousapps.jvm.projects.JavaToolchainProject
+import org.gradle.testkit.runner.GradleRunner
+
+import static com.google.common.truth.Truth.assertThat
+import static com.autonomousapps.utils.Runner.build
+import static com.autonomousapps.utils.Runner.runner
+
+final class JavaToolchainSpec extends AbstractJvmSpec {
+  def "does not fail with Java #javaToolchainVersion toolchain (#gradleVersion)"() {
+    given: "a Gradle project using Java toolchain version #javaToolchainVersion that's been built"
+    def project = new JavaToolchainProject(javaToolchainVersion)
+    gradleProject = project.gradleProject
+    build(gradleVersion, gradleProject.rootDir, 'tasks', '--all')
+    build(gradleVersion, gradleProject.rootDir, 'build')
+
+    when: 'running buildHealth'
+    // TODO This passes because it is not running abiAnalysisMain nor explodeByteCodeSourceMain which are the problem
+    // I added tasks --all above to see and indeed they're not part of buildHealth like in the /poc folder I made
```

Comment:
@autonomousapps At the moment this is where I need your wisdom the most. How can I make the tasks `abiAnalysisMain` and `explodeByteCodeSourceMain` be part of the project. Those are the two that actually break right now. As you can see in the poc I added I didn't really "do" anything special to get them into the project so I'm confused why they aren't being added here.

Reply:
Never mind, I figured it out. I needed to apply java lib plugin instead of java plugin.